core/src/utilities/helpers/generic-helpers.js

```diff
@@ -140,6 +140,9 @@ class GenericHelpersClass {
    * @returns {string} string without leading slash
    */
   trimLeadingSlash(str) {
+    if (typeof str !== 'string') {
+      return '';
+    }
```

Comment:
I would suggest having a generic function like
`const isString = (value) => typeof value === 'string' || value instanceof String;`

Then we could reuse it here, also for `trimTrailingSlash` function